### PR TITLE
adding additional example to Tables doc

### PR DESCRIPTION
**Pull request by SaraOgaz**
Hi,

I wanted to add an extra example to the Tables doc that shows how to filter rows on a ``Table`` using a boolean mask (conditionals).  I couldn't find an example of this anywhere else except for one that was actually using the filtering to replace values in the table.  But many beginning users may not see this example if they weren't looking in the "modifying a table" section.

### CODE REVIEW

**Comment by pllim** [comment]
I milestoned to 3.1 because it is enhancement but feel free to remilestone as needed.

**Comment by bsipocz** [comment]
@pllim - Doc enhancements are fine to be backported, given that they work with the bugfix/LTS version (as this one does).

File: docs/table/access_table.rst [hunk 1]
```diff
@@ -288,6 +288,19 @@ selected rows or columns.  ::
        9.000  11
       12.000  14
 
+We can select rows from a table using conditionals to create boolean masks. A
+table indexed with a boolean array will only return rows where the mask array
+is equal to `True`. Different conditionals can be combined using the bitwise
```
Comment (taldcroft) [inline docs/table/access_table.rst]: Suggest writing as below.  Slightly fussy, but by analogy with PEP8 one writes `is True` not `is equal to True`.
```
element is `True`.  Different ...
```

File: docs/table/access_table.rst [hunk 2]
```diff
@@ -288,6 +288,19 @@ selected rows or columns.  ::
...
+operators.  ::
+
+  >>> print(t[(t['a'] > 4) & (t['b'] > 8)])  # Table rows where column a > 4
```
Comment (taldcroft) [inline docs/table/access_table.rst]: This expression is a little dense.  For clarity I would suggest writing this as:
```
>>> mask = (t['a'] > 4) & (t['b'] > 8)  # Table rows where column a > 4 and b > 8
>>> print(t[mask])
```
I actually do this separation most of the time in my production code.

**Comment by SaraOgaz** [comment]
ahhh, something crazy happened when i was squashing... need to figure out how to fix that, maybe another rebase will do it.

**Comment by taldcroft** [comment]
@SaraOgaz - the text looks fine now, thanks!  I haven't dug into all the circleci fails.  Can you do so and confirm they are unrelated?  If so I'll approve and merge.

**Comment by SaraOgaz** [comment]
@taldcroft, there was a bunch of stuff also failing at the time that these ran, could someone with permission reset them? If not, I could make some kind of bum commit I guess.

**Comment by SaraOgaz** [comment]
whoo @taldcroft, I rebased, looks like all test are cleared now.

**Comment by taldcroft** [comment]
Thanks @SaraOgaz !

**Pull request merged**